src/Operations.ts

```diff
@@ -309,6 +309,74 @@ export class Operations {
     }
   }
 
+  public setRowOrder(sheetId: number, rowMapping: [number, number][]): void {
+    const buffer: [SimpleCellAddress, ClipboardCell][][] = []
+    for(const [source, target] of rowMapping ) {
+      if(source!==target) {
+        const rowRange = AbsoluteCellRange.spanFrom({sheet: sheetId, col: 0, row: source}, Infinity, 1)
+        if (this.dependencyGraph.matrixMapping.isFormulaMatrixInRange(rowRange)) {
+          throw new SourceLocationHasMatrixError()
+        }
+        this.dependencyGraph.breakNumericMatricesInRange(rowRange)
+        const row = this.getRangeClipboardCells(rowRange)
+        buffer.push(
+          row.map(
+            ([{sheet, col, row}, cell]) => [{sheet, col, row: target}, cell]
+          )
+        )
+      }
+    }
+    buffer.forEach(
+      row => this.restoreClipboardCells(sheetId, row.values())
+    )
+  }
+
+  public testRowOrderForMatrices(sheetId: number, rowMapping: [number, number][]): boolean {
+    for(const [source, target] of rowMapping ) {
+      if(source!==target) {
+        const rowRange = AbsoluteCellRange.spanFrom({sheet: sheetId, col: 0, row: source}, Infinity, 1)
+        if (this.dependencyGraph.matrixMapping.isFormulaMatrixInRange(rowRange)) {
+          return false
+        }
+      }
+    }
+    return true
+  }
+
+  public setColumnOrder(sheetId: number, columnMapping: [number, number][]): void {
+    const buffer: [SimpleCellAddress, ClipboardCell][][] = []
+    for(const [source, target] of columnMapping ) {
+      if(source!==target) {
+        const rowRange = AbsoluteCellRange.spanFrom({sheet: sheetId, col: source, row: 0}, 1, Infinity)
+        if (this.dependencyGraph.matrixMapping.isFormulaMatrixInRange(rowRange)) {
+          throw new SourceLocationHasMatrixError()
+        }
+        this.dependencyGraph.breakNumericMatricesInRange(rowRange)
+        const column = this.getRangeClipboardCells(rowRange)
+        buffer.push(
+          column.map(
+            ([{sheet, col, row}, cell]) => [{sheet, col: target, row}, cell]
+          )
+        )
+      }
+    }
+    buffer.forEach(
+      column => this.restoreClipboardCells(sheetId, column.values())
+    )
+  }
+
+  public testColumnOrderForMatrices(sheetId: number, columnMapping: [number, number][]): boolean {
```

Comment:
It seems that this method (and `testRowOrderForMatrices`) can be placed directly in `CrudOperations`.